plugin-server/src/worker/ingestion/team-manager.ts

```diff
@@ -50,26 +49,41 @@ export class TeamManager {
     }
 
     public async getTeamByToken(token: string): Promise<Team | null> {
+        /**
+         * Validates and resolves the api token from an incoming event.
+         *
+         * Caching is added to reduce the load on Postgres, not to be resilient
+         * to failures. If PG is unavailable, this function will trow and the
+         * lookup must be retried later.
+         *
+         * Returns null if the token is invalid.
+         */
+
         const cachedTeamId = this.tokenToTeamIdCache.get(token)
 
-        // tokenToTeamIdCache.get returns `undefined` if the value doesn't
-        // exist so we check for the value being `null` as that means we've
-        // explictly cached that the team does not exist
```

Comment:
I liked this comment a bit better than simply `Negative lookups (`null` instead of `undefined`)` - note that we had an problem because we messed this up somewhere else for a while (group properties were not added to events)